src/main/java/com/syncapse/jenkinsci/plugins/awscloudformationwrapper/CloudFormationBuildStep.java

```diff
@@ -109,13 +114,39 @@ protected CloudFormation newCloudFormation(PostBuildStackBean postBuildStackBean
 			recipe = build.getWorkspace().child(postBuildStackBean.getCloudFormationRecipe()).readToString();
 		}
 
+		List<Parameter> mergedParameters = Collections.emptyList();
+		Map<String, String> parsedParameters = postBuildStackBean.getParsedParameters(env);
+		if (parsedParameters != null) {
+			mergedParameters = new ParametersConverter().convert(parsedParameters);
+		}
+
+		if (postBuildStackBean.getParametersFile() != null && !postBuildStackBean.getParametersFile().isEmpty()) {
+			String parametersJsonString = build.getWorkspace().child(postBuildStackBean.getParametersFile()).readToString();
+			List<Parameter> fileParameters = new ParametersParser().parse(parametersJsonString);
+
+			mergedParameters = mergeParameters(mergedParameters, fileParameters);
+		}
+
 		return new CloudFormation(logger, postBuildStackBean.getStackName(), isURL,
-				recipe, postBuildStackBean.getParsedParameters(env),
+				recipe, mergedParameters,
 				postBuildStackBean.getTimeout(), postBuildStackBean.getParsedAwsAccessKey(env),
 				postBuildStackBean.getParsedAwsSecretKey(env),
 				postBuildStackBean.getAwsRegion(), env,false,postBuildStackBean.getSleep());
 
 	}
+
+	private List<Parameter> mergeParameters(List<Parameter> first, List<Parameter> second) {
```

Comment:
The mergeParameters method seems like something that should naturally go in the ParametersConverter class, or take all of the method and build a ParameterUtils class that handles merging, parsing, converting, etc.